### PR TITLE
feat(action plans): monthly action plan form, men-84

### PR DESCRIPTION
**Pull request by jimburch**
Navigate to "/action-plan"

This is the form portions of the action plans feature. Right now it's a standalone page. Next PR will prompt the user every month to fill out a new form

### CODE REVIEW

File: src/views/ActionPlan/index.tsx [hunk 1]
```diff
@@ -0,0 +1,448 @@
...
+type ActionPlanProps = {
+  currentUser: CurrentUser;
+};
+
+const ActionPlan = ({ currentUser }: ActionPlanProps): JSX.Element => {
+  const classes = useStyles();
+  const { enqueueSnackbar } = useSnackbar();
+  const navigate = useNavigate();
+
+  const handleSubmit = async (values: any) => {
```
Comment (erwstout) [inline src/views/ActionPlan/index.tsx:62]: got an `any` here, lets type that badboi

File: src/views/ActionPlan/index.tsx [hunk 2]
```diff
@@ -0,0 +1,448 @@
...
+    const keyActionItems = [
+      values.keyActionItem1,
+      values.keyActionItem2,
+      values.keyActionItem3,
+      values.keyActionItem4,
+      values.keyActionItem5,
+    ];
```
Comment (erwstout) [inline src/views/ActionPlan/index.tsx:69]: can we just spread these values? `[...values]`

File: src/views/ActionPlan/index.tsx [hunk 3]
```diff
@@ -0,0 +1,448 @@
...
+
+    await axios
+      .post(`${envConfig.API_URL}/v1/action-plans`, {
```
Comment (erwstout) [inline src/views/ActionPlan/index.tsx:72]: i think we can pass the types to axios here, right? `.post<ActionPlan>`

**Review by erwstout — APPROVED** [review]
some comments, but otherwise g2g

File: src/views/ActionPlan/index.tsx (hunk 2 above)
Comment (jimburch) [reply src/views/ActionPlan/index.tsx:69]: It's not all the values, just the 5 key action times

**Pull request merged**